pkg/controllers/management/auth/project_cluster_handler.go

```diff
@@ -212,40 +246,100 @@ func (m *mgr) reconcileCreatorRTB(obj runtime.Object) (runtime.Object, error) {
 			return obj, nil
 		}
 
-		rtbName := "creator"
-		om := v1.ObjectMeta{
-			Name:      rtbName,
-			Namespace: metaAccessor.GetName(),
+		roleJSON, ok := metaAccessor.GetAnnotations()[roleTemplatesRequired]
+		if !ok {
+			logrus.Warnf("%v %v has no creator-role-bindings annotation. Cannot create role bindings", typeAccessor.GetKind(), metaAccessor.GetName())
+			return obj, nil
+		}
+
+		roleMap := make(map[string][]string)
+		err = json.Unmarshal([]byte(roleJSON), &roleMap)
+		if err != nil {
+			return obj, err
 		}
 
 		switch typeAccessor.GetKind() {
 		case v3.ProjectGroupVersionKind.Kind:
-			if rtb, _ := m.prtbLister.Get(metaAccessor.GetName(), rtbName); rtb != nil {
-				return obj, nil
+			for _, role := range roleMap["required"] {
+				rtbName := "creator-" + role
+
+				if rtb, _ := m.prtbLister.Get(metaAccessor.GetName(), rtbName); rtb != nil {
```

Comment:
per our discussion, need to change to checking the created json map, so that we dont continually recreate rolebindings for the creator if they are being legitimately deleted